src/main/java/com/spotify/docker/BuildMojo.java

```diff
@@ -198,6 +202,20 @@ public boolean getForceTags() {
   }
 
   @Override
+  public void execute() throws MojoExecutionException
```

Comment:
instead of adding the `LOCK`, I think you could also just mark this method as `synchronized`.